### PR TITLE
add collection code to set

### PR DESCRIPTION
**Pull request by ozym**
This code joins the dataloggers, the sensors, and the streams for the times when they overlap.  There is also provision for times which may need adjustments (i.e. gains, polarities) which may not relate to the physical connections between the various parts.

This is the engine for building stationxml like meta data.

### CODE REVIEW

File: meta/collection.go [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+						}
+						if !span.Overlaps(c.Span) {
+							continue
+						}
+						dataloggers = append(dataloggers, c)
+					}
+					sort.Slice(dataloggers, func(i, j int) bool {
+						return dataloggers[i].Span.Start.Before(dataloggers[j].Span.Start)
+					})
+
```
Comment (bpeng) [inline meta/collection.go:226]: don't know if I fully understand the logic, but seems the `gains`, `sensors` and `dataloggers` are not dependent on `channels`, and can be calculated outside the channels loop L155

File: meta/collection.go [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+									continue
+								}
+								if !span.Overlaps(c.Span) {
+									continue
+								}
+								sensors = append(sensors, c)
+							}
+							sort.Slice(sensors, func(i, j int) bool {
+								return sensors[i].Span.Start.Before(sensors[j].Span.Start)
+							})
```
Comment (bpeng) [inline meta/collection.go:361]: similar to the above, check if it is possible to calculate `gains` and `sensors` out side the channels loop

**Review by bpeng — CHANGES_REQUESTED** [review]
Code looks tidy, little check

**Comment by ozym** [comment]
@bpeng I did a recheck and found there should be an extra constraint for the sensor/datalogger calibrations for the Number. These files should have the Number explicitly given (as it's a calibration).

I've also adjusted the order of the loops to make the time span get evaluated as soon as possible, and to be consistent for both sets.

**Review by bpeng — APPROVED** [review]
OK

**Pull request merged**